### ISSUE TITLE
How to get ref to the input?

### ISSUE THREAD

**Issue by mbrevda** [s-yadav/react-number-format#125]
I'm using this with material-ui like:

```js

class Amount extends Component {
  render() {
    return <NumberFormat {...this.props} decimalScale={2} />
  }
}

render() {
            <TextField
              className={classes.textField}
              fullWidth={true}
              label="Amount"
              name="amount"
              onChange={this.onAmountChange}
              required={true}
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">$</InputAdornment>
                ),
                inputRef: ref => this.trackRef('amount', ref),
                inputComponent: Amount
              }}
}
```

But I don't get a ref to the actual input. How can I get the ref?

Edit: it seem that the `omit` help function isn't copying a `ref` function if one is provided

**Comment by s-yadav** [comment]
`ref` attributes are not passed as props to the component See (https://reactjs.org/warnings/special-props.html). They are internally used by react.

I guess in your Amount component you have to manually pass ref . Something like
```jsx
<NumberFormat ref={this.props. inputRef} {...resr} decimalScale={2} />
```
But in this case as well ref will not forward to actual input. I guess I also have to support something similar `inputRef` like TextField Component.

**Comment by mbrevda** [comment]
>  I guess I also have to support something similar inputRef like TextField Component

Cool, thanks!

**Comment by pvpacheco** [comment]
I've just submitted a pull request (#131) implementing the inputRef prop, hope it helps...

**Comment by s-yadav** [comment]
This has been implemented on v3.1.0
https://github.com/s-yadav/react-number-format#getting-reference

**Comment by mbrevda** [comment]
Whoa! This is awesome, thanks!
